src/mock.rs

```diff
@@ -0,0 +1,151 @@
+use crate::expectation::model::{Expectation, HttpMethod};
+
+pub struct IncomingRequest {
+    pub method: HttpMethod,
+    pub code: i32,
+    pub path: String,
+}
+
+pub fn look_for_expectation(
+    expectations: &Vec<Expectation>,
+    req: IncomingRequest,
+) -> Option<&Expectation> {
+    expectations.iter().find(|e| {
+        let match_method = e.request.method.map(|m| m == req.method).unwrap_or(true);
+
+        let match_path = e
+            .request
+            .path
+            .clone()
+            .map(|p| p == req.path)
+            .unwrap_or(true);
+
+        match_method && match_path
+    })
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use crate::expectation::model::{HttpRequest, HttpResponse};
+
+    #[test]
+    fn test_accept_matching_method() {
+        let req = HttpRequest {
+            method: Some(HttpMethod::GET),
+            path: None,
+        };
+
+        let resp = HttpResponse {
+            status_code: Some(200),
+            status_reason: None,
+            body: None,
+        };
+
+        let exp = Expectation {
+            request: req,
+            response: resp,
+        };
+
+        let income = IncomingRequest {
+            method: HttpMethod::GET,
+            code: 200,
+            path: String::from("/foo/bar"),
+        };
+
+        let v = vec![exp.clone()];
+        let tested = look_for_expectation(&v, income);
+
+        assert_eq!(Some(&exp), tested);
+    }
+
+    #[test]
+    fn test_refuse_wrong_method() {
+        let req = HttpRequest {
+            method: Some(HttpMethod::POST),
+            path: None,
+        };
+
+        let resp = HttpResponse {
+            status_code: Some(200),
+            status_reason: None,
+            body: None,
+        };
+
+        let exp = Expectation {
+            request: req,
+            response: resp,
+        };
+
+        let income = IncomingRequest {
+            method: HttpMethod::GET,
+            code: 200,
+            path: String::from("/foo/bar"),
+        };
+
+        let v = vec![exp.clone()];
+        let tested = look_for_expectation(&v, income);
+
+        assert_eq!(None, tested);
+    }
+
+    #[test]
+    fn test_accept_matching_path() {
+        let req = HttpRequest {
+            method: Some(HttpMethod::GET),
+            path: Some(String::from("/foo/bar")),
+        };
+
+        let resp = HttpResponse {
+            status_code: Some(200),
+            status_reason: None,
+            body: None,
+        };
+
+        let exp = Expectation {
+            request: req,
+            response: resp,
+        };
+
+        let income = IncomingRequest {
+            method: HttpMethod::GET,
+            code: 200,
+            path: String::from("/foo/bar"),
+        };
+
+        let v = vec![exp.clone()];
+        let tested = look_for_expectation(&v, income);
+
+        assert_eq!(Some(&exp), tested);
```

Comment:
Could you add another expectation with a request with only `method: Some(HttpMethod::GET),` configured to check if it take the one with the more parameters ?

Reply:
I propose to open another MR (and discussion?) to talk about expectation strategy. I want to keep this MR really straightforward : take the first expectation in the list that match the request and go on.

Reply:
Sure good idea
Let's keep the scope of the PR small :+1:

Reply:
I opened the issue #24